core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -363,7 +363,10 @@ where
         const MAX_TRANSACTION_CHECKS: usize = 10_000;
         let mut transaction_ids = Vec::with_capacity(MAX_TRANSACTION_CHECKS);
 
-        while let Some(id) = self.container.pop() {
+        while transaction_ids.len() < MAX_TRANSACTION_CHECKS {
+            let Some(id) = self.container.pop() else {
+                break;
```

Comment:
not sure why this fmt was different on master - need a `;` in the backport. added in 7970a43f90